File: backend/spring/src/main/java/nidhal/stackoverflowstatus/entity/Question.java
package nidhal.stackoverflowstatus.entity;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;
import java.util.Objects;

@Builder
@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Question implements Serializable {
    @Id
    private long question_id;
    public Boolean is_answered;
    // Adjust the length of the column to avoid the following error:
    // Data truncation: Data too long for column 'title' at row 1
    @Column(length = 510)
    private String title;
    @Column(length = 510)
    private String link;
    private int score;
    private int answer_count;
    public long accepted_answer_id;
    private int view_count;
    @ElementCollection
    private List<String> tags;
    private long creation_date;
    public long closed_date;
    public long last_activity_date;
    public long last_edit_date;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Question question)) return false;
        return getQuestion_id() == question.getQuestion_id() &&
               getScore() == question.getScore() &&
               getAnswer_count() == question.getAnswer_count() &&
               getAccepted_answer_id() == question.getAccepted_answer_id() &&
               getView_count() == question.getView_count() &&
               getCreation_date() == question.getCreation_date() &&
               getClosed_date() == question.getClosed_date() &&
               getLast_activity_date() == question.getLast_activity_date() &&
               getLast_edit_date() == question.getLast_edit_date() &&
               Objects.equals(getIs_answered(), question.getIs_answered()) &&
               Objects.equals(getTitle(), question.getTitle()) &&
               Objects.equals(getLink(), question.getLink()) &&
               Objects.equals(getTags(), question.getTags());
    }

}

File: backend/dotnet/stackoverflow-statistics/Models/Question.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace stackoverflow_statistics.Models;

public class Question
{
    [Key]
    public long QuestionId { get; init; }

    public bool? IsAnswered { get; set; }

    [MaxLength(510)]
    public string Title { get; init; }

    [MaxLength(510)]
    public string Link { get; init; }

    public int Score { get; init; }

    public int AnswerCount { get; init; }

    public long AcceptedAnswerId { get; init; }

    public int ViewCount { get; init; }

    [NotMapped]
    public List<string> Tags { get; init; }

    public long CreationDate { get; init; }

    public long ClosedDate { get; init; }

    public long LastActivityDate { get; init; }

    public long LastEditDate { get; init; }

}